wit/couchbase/types.wit

```diff
@@ -0,0 +1,219 @@
+  /// - a valid JSON string (which may or may not be an object)
+  /// - an implementer-controlled value which is a valid JSON object (as defined by the implementer)
+  ///
+  /// The implementer-controlled value can be any implementation (usually a more efficient one)
+  /// that can be understood by the implementer. One requirement on this value is that it must be
+  /// possible to convert it to/from a raw JSON string at any time.
+  ///
+  /// An opaque resource that represents a JSON value, and can be efficiently
+  /// manipulated by implementers.
+  variant document {
+    /// A stringified JSON value represented by a resource, often used by callers
+    /// that cannot or do not want to construct an efficient implementer specific JSON value
+    raw(json-string),
+
+    /// A JSON value represented by a more efficient but opaque implementer-specific representation
+    %resource(document-value),
+  }
+
+  /// Errors related to bucket usage
+  variant bucket-error {
+    /// A completely unexpected error
+    unexpected(string)
+  }
+
+  /// Metadata related to any mutation on a Couchbase collection (ex. CRUD operations)
+  record mutation-metadata {
+    /// CAS revision of the document
+    cas: u64,
+    /// The bucket on which the mutation was performed
+    bucket: string,
+    /// The ID of the vbucket (partition) that the operation was performed on
+    partition-id: u64,
+    /// The UUID of the vbucket (partition) that the operation was performed on
+    partition-uuid: u64,
+    /// The sequence number of the operation performed on the vbucket (partition)
+    seq: u64,
+  }
+
+  /// Durability level that should be used for operations
+  enum durability-level {
+    unknown,
+
+    none,
+
+    /// Replicate (hold in memory) to a majority of nodes
+    replicate-majority,
+
+    /// Replicate (hold in memory) to a majority of nodes, persist (write to disk) to master
+    replicate-majority-persist-master,
+
+    /// Persist to a majority of nodes
+    persist-majority,
+  }
+
+  /// As functions cannot be represented as part of types in WIT,
+  /// we represent static retry strategies
+  variant retry-strategy {
+    /// Retry a certain number of times with a given interval between each retry)
+    interval-times-ms(tuple<u64, u64>),
+  }
+
+  /// Level of data consistency required for a query
+  enum query-scan-consistency {
+    /// No data consistency required
+    not-bounded,
+    /// Request-level data consistency is required
+    request-plus,
+  }
+
+  /// Profiling mode to use during a query
+  enum query-profile-mode {
+    /// Disable query profiling
+    none,
+    /// Include phase-boudned profiling information in the response
+    phases,
+    /// Include timing profiling informatoin in the query response
+    timings,
+  }
+
+  /// Level of data consistency required for a search
+  enum search-scan-consistency {
+    /// No data consistency required
+    not-bounded,
```

Comment:
Didn't you forgot `request-plus` ?

Reply:
I don't think so? https://pkg.go.dev/github.com/couchbase/gocb/v2#SearchScanConsistency only contains one value. I think you're thinking of https://github.com/couchbase/gocb/blob/v2.9.0/query_options.go .

I'm fairly sure I'm reading the golang right -- there's only one value there.

Reply:
You're right ! I'm not sure if the forgot it in the SDKs or if it's just not implemented yet 🧐

I've asked on the forum : https://www.couchbase.com/forums/t/search-with-requestplus/38703

Reply:
Oh yeah, and note that this is version 0.1.0-draft! We'll have plenty of time to update and change the contract (and hit 0.1.0 in the first place)!

Reply:
Of course !